floors/rule.go

```diff
@@ -0,0 +1,382 @@
+package floors
+
+import (
+	"encoding/json"
+	"fmt"
+	"math/bits"
+	"regexp"
+	"sort"
+	"strings"
+
+	"github.com/buger/jsonparser"
+	"github.com/mxmCherry/openrtb/v16/openrtb2"
+	"github.com/prebid/prebid-server/currency"
+	"github.com/prebid/prebid-server/openrtb_ext"
+)
+
+const (
+	SiteDomain string = "siteDomain"
+	PubDomain  string = "pubDomain"
+	Domain     string = "domain"
+	Bundle     string = "bundle"
+	Channel    string = "channel"
+	MediaType  string = "mediaType"
+	Size       string = "size"
+	GptSlot    string = "gptSlot"
+	PbAdSlot   string = "pbAdSlot"
+	Country    string = "country"
+	DeviceType string = "deviceType"
+	Tablet     string = "tablet"
+	Phone      string = "phone"
+)
+
+func getFloorCurrency(floorExt *openrtb_ext.PriceFloorRules) string {
+	floorCur := "USD"
+	if floorExt.Data.Currency != "" {
+		floorCur = floorExt.Data.Currency
+	}
+
+	if floorExt.Data.ModelGroups[0].Currency != "" {
+		floorCur = floorExt.Data.ModelGroups[0].Currency
+	}
+	return floorCur
+}
```

Comment:
Looks like this could be used to help resolve my earlier comment about USD currency.

Reply:
Earlier comment was in Enforcement and this logic is for getting currency from floor json